File: 500.java
class Solution {
    public String[] findWords(String[] words) {
        // time: O(n * k), space: O(1)
        List<String> res = new ArrayList<>();
        String[] rows = {"qwertyuiop", "asdfghjkl", "zxcvbnm"};
        for (String w : words) {
            String lower = w.toLowerCase();
            for (String r : rows) {
                if (isValid(r, lower)) {
                    res.add(w);
                }
            }
        }
        return res.toArray(new String[0]);
    }

    private boolean isValid(String r, String w) {
        for (char c : w.toCharArray()) {
            if (r.indexOf(c) == -1) {
                return false;
            }
        }
        return true;
    }
}

File: 500.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        res = []
        for w in words:
            if set("qwertyuiop") >= set(w.lower()) or set("asdfhgjkl") >= set(w.lower()) or set("zxcvbnm") >= set(w.lower()):
                res.append(w)
        return res